### PR TITLE
fix cascading backoff

### PR DESCRIPTION
**Pull request by freehan**
#20550

### CODE REVIEW

**Comment by dchen1107** [comment]
lgtm

File: pkg/kubelet/container/image_puller.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func (puller *imagePuller) PullImage(pod *api.Pod, container *api.Container, pul
 		}
 	}
 
-	backOffKey := fmt.Sprintf("%s_%s", pod.Name, container.Image)
+	backOffKey := fmt.Sprintf("%s_%s_%s", pod.Name, pod.Namespace, container.Image)
```
Comment (yujuhong) [inline pkg/kubelet/container/image_puller.go]: nit: use pod UID to avoid hitting the case of pod being recreated with the same name/namespace.

**Comment by freehan** [comment]
can I get a lgtm on this?

**Comment by ArtfulCoder** [comment]
merging PR to help with test flakiness

**Pull request merged**